File: examples/test_list_append.py
"""Test list append method"""

class Logger:
    def __init__(self):
        self.messages = []
    
    def log(self, msg: str) -> int:
        """Add a message to the log"""
        self.messages.append(msg)
        return len(self.messages)

def test_logger():
    logger = Logger()
    count = logger.log("Hello")
    return count

File: examples/test_list_append.rs
use serde_json;
#[derive(Debug, Clone)]
pub struct Logger {
    pub messages: Vec<serde_json::Value>,
}
impl Logger {
    pub fn new() -> Self {
        Self {
            messages: Vec::new(),
        }
    }
    pub fn log(&self, msg: String) -> i32 {
        self.messages.push(msg);
        return self.messages.len();
    }
}
#[doc = " Depyler: verified panic-free"]
#[doc = " Depyler: proven to terminate"]
pub fn test_logger() {
    let logger = Logger::new();
    let count = logger.log("Hello".to_string());
    count
}
